feature/signupflow/entercode/src/main/java/com/epmedu/animeal/signup/entercode/data/EnterCodeRepositoryImpl.kt

```
@@ -2,21 +2,36 @@ package com.epmedu.animeal.signup.entercode.data
 
 import androidx.datastore.core.DataStore
 import androidx.datastore.preferences.core.Preferences
+import com.epmedu.animeal.auth.AuthAPI
+import com.epmedu.animeal.auth.AuthRequestHandler
 import com.epmedu.animeal.common.constants.DataStorePreferencesKey.phoneNumberKey
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 internal class EnterCodeRepositoryImpl @Inject constructor(
-    dataStore: DataStore<Preferences>
+    dataStore: DataStore<Preferences>,
+    private val authAPI: AuthAPI,
 ) : EnterCodeRepository {
 
     override val phoneNumber: Flow<String> = dataStore.data
         .map { preferences ->
             preferences[phoneNumberKey] ?: PHONE_NUMBER_PLACEHOLDER
```

This is used in sendCode, won't it send code to default number if phoneNumber is not found in preferences ?

---

Actually we shouldn't open this screen, if we didn't save a phone number. I guess we can throw an exception to prevent invalid state of the app

---

Just let it be an empty string as default. If we make mistake in future we won't be sending sms to a random mobile number.

---

Let's use empty string instead